roles/openshift_aws/tasks/setup_master_group.yml

```diff
@@ -26,7 +26,7 @@
     groups: "{{ openshift_aws_masters_groups }}"
     name: "{{ item.public_dns_name }}"
     hostname: "{{ openshift_aws_clusterid }}-master-{{ item.instance_id[:-5] }}"
-    openshift_node_group_name: "{{ openshift_aws_node_group_mappings[item.tags['sub-host-type']] }}"
```

Comment:
We are doing something similar to original way in gcp now: https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_gcp/tasks/setup_scale_group_facts.yml#L41

I created a simple dictionary to map group types to node-groups.  I changed the order in which hosts are added to the respective groups; 1st add everything to nodes, set type to compute; next, add hosts to infra, set type to infra, finally hosts to master as type master;

This would prevent needing to tag/relabel hosts in cloud provider.